cmd/boulder-publisher/main.go

```diff
@@ -19,8 +24,26 @@ const clientName = "Publisher"
 func main() {
 	app := cmd.NewAppShell("boulder-publisher", "Submits issued certificates to CT logs")
 	app.Action = func(c cmd.Config, stats statsd.Statter, auditlogger *blog.AuditLogger) {
-		pubi, err := publisher.NewPublisherImpl(c.Common.CT)
-		cmd.FailOnError(err, "Could not setup Publisher")
+		logs := make([]*publisher.Log, len(c.Common.CT.Logs))
+		var err error
+		for i, ld := range c.Common.CT.Logs {
+			fmt.Println(ld)
+			logs[i], err = publisher.NewLog(ld.URI, ld.PublicKey)
+			cmd.FailOnError(err, "Unable to parse CT log description")
+		}
+
+		if c.Common.CT.IntermediateBundleFilename == "" {
+			auditlogger.Err("No CT submission bundle provided")
+			os.Exit(1)
```

Comment:
`cmd.FailOnError` requires `err != nil` to trigger, so the replacement would end up being `cmd.FailOnError(fmt.Errorf(...), ...)` which itself isn't a great pattern. Instead I just went for what we do inside `FailOnError` since it seems like the least horrible option. As far as I remember @jsha agreed out-of-band.